source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/Calculation/CalculationOrchestrationStateTests.cs

```diff
@@ -0,0 +1,205 @@
+using Energinet.DataHub.Wholesale.Orchestrations.IntegrationTests.DurableTask;
+using Energinet.DataHub.Wholesale.Orchestrations.IntegrationTests.Extensions;
+using Energinet.DataHub.Wholesale.Orchestrations.IntegrationTests.Fixtures;
+using FluentAssertions;
+using Xunit.Abstractions;
+
+namespace Energinet.DataHub.Wholesale.Orchestrations.IntegrationTests.Functions.Calculation;
+
+[Collection(nameof(OrchestrationsAppCollectionFixture))]
+public class CalculationOrchestrationStateTests : IAsyncLifetime
+{
+    public CalculationOrchestrationStateTests(
+        OrchestrationsAppFixture fixture,
+        ITestOutputHelper testOutputHelper)
+    {
+        Fixture = fixture;
+        Fixture.SetTestOutputHelper(testOutputHelper);
+    }
+
+    private OrchestrationsAppFixture Fixture { get; }
+
+    public Task InitializeAsync()
+    {
+        Fixture.EnsureAppHostUsesMockedDatabricksJobs();
+        Fixture.AppHostManager.ClearHostLog();
+
+        // Clear mappings etc. before each test
+        Fixture.MockServer.Reset();
+
+        Fixture.ServiceBusListenerMock.ResetMessageHandlersAndReceivedMessages();
+
+        return Task.CompletedTask;
+    }
+
+    public Task DisposeAsync()
+    {
+        Fixture.SetTestOutputHelper(null!);
+
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Verifies that:
+    ///  - The orchestration can complete a full run.
+    ///  - The calculation state is updated as expected.
+    ///  - The orchestrator completes with the expected output.
+    ///  - A service bus message is sent as expected.
+    /// </summary>
+    [Fact]
+    public async Task GivenExpectedCalculationFlow_WhenHandlingCalculationOrchestration_OrchestrationCompletesWithExpectedStateUpdatesAndServiceBusMessage()
+    {
+        // Arrange
+        var dbContext = Fixture.DatabaseManager.CreateDbContext();
+
+        // => Databricks Jobs API
+        var jobId = Random.Shared.Next(1, 1000);
+        var runId = Random.Shared.Next(1000, 2000);
+
+        // The current databrick calculation state. Can be null, "PENDING", "RUNNING", "TERMINATED" (success)
+        // The mock response will wait for the value to not be null before returning
+        string? currentCalculationJobState = null;
+        Fixture.MockServer
+            .MockJobsList(jobId)
+            .MockJobsGet(jobId)
+            .MockJobsRunNow(runId)
+            // ReSharper disable once AccessToModifiedClosure -- We need to modify calculation job state in outer scope
+            .MockJobsRunsGet(runId, () => currentCalculationJobState);
+
+        // => Databricks SQL Statement API
+        var chunkIndex = 0;
+        var statementId = Guid.NewGuid().ToString();
+        var path = "GetDatabricksDataPath";
+
+        // This is the calculationId returned in the energyResult from the mocked databricks.
+        // It should match the ID returned by the http client calling 'api/StartCalculation'.
```

Comment:
```suggestion
        // It should match the ID returned by the http client calling 'api/StartCalculation'.
```
The test will not fail if this is not true. But it this statement should be true if we want to memic a "proper" scenario